x/programs/rust/wasmlanche-sdk/src/memory.rs

```diff
@@ -81,7 +81,7 @@ pub extern "C" fn alloc(len: usize) -> HostPtr {
 
     ALLOCATIONS.with_borrow_mut(|s| s.insert(ptr, len));
 
-    HostPtr(ptr.cast_const())
+    HostPtr(ptr)
```

Comment:
Closing this since the others changes have been incorporated as part of rewrites. The only missing thing is this redundant `cast_const` call